File: src/Models/Request/SimClassRequestUpdate.cs
using System.ComponentModel.DataAnnotations;
using System.Diagnostics.CodeAnalysis;
using System.Text.Json.Serialization;
using Models.Enums;

namespace Models.Request;

[ExcludeFromCodeCoverage]
public class SimClassRequestUpdate()
{
    [Required(ErrorMessage = "Id is a Guid and it is required.")]
    public string Id { get; set; } = string.Empty;

    [JsonIgnore]
    public Guid IdClass => Guid.TryParse(Id, out var guid) ? guid : Guid.Empty;

    [Required(ErrorMessage = "Name is required.")]
    public string Name { get; set; } = string.Empty;

    [JsonConverter(typeof(JsonStringEnumConverter))]
    [Required(ErrorMessage = "State is required and their values should be: Normal, Abstract, Sealed.")]
    public SimModelsAccesibility? State { get; set; }

    [JsonIgnore]
    public Guid BaseClassId => Guid.TryParse(IdBaseClass, out var guid) ? guid : Guid.Empty;

    public string IdBaseClass { get; set; } = "11111111-1111-1111-1111-111111111111";
    public List<MethodRequest> Methods { get; set; } = [];
    public List<AttributeRequest> Attributes { get; set; } = [];
}

File: frontend/src/app/models/request/SimClassRequestUpdate.ts
import { MethodRequest } from './MethodRequest';
import { AttributeRequest } from './AttributeRequest';
import { InterfaceRequestUpdate } from './InterfaceRequestUpdate';

export class SimClassRequestUpdate {
    id: string = '';
    name: string = '';
    state: string = 'Normal'; // 'Normal' | 'Abstract' | 'Sealed'
    idBaseClass: string = '11111111-1111-1111-1111-111111111111';
    methods: MethodRequest[] = [];
    attributes: AttributeRequest[] = [];
    implements: InterfaceRequestUpdate[] = [];
    idBaseNamespace: string = '';

    toString(): string {
        return `SimClassRequestUpdate(id=${this.id}, name=${this.name}, state=${this.state}, methods=${this.methods.length}, attributes=${this.attributes.length})`;
    }
}